controllers/apply-next/simple/fields.js

```diff
@@ -538,39 +573,58 @@ const allFields = {
             cy: ''
         },
         type: 'radio',
-        options: legalContactRoles,
-        isRequired: true,
-        get schema() {
-            return Joi.array()
-                .items(Joi.string().valid(this.options.map(option => option.value)))
-                .single()
-                .optional();
+        options(formData) {
```

Comment:
I've allowed options to (optionally) be a function to allow us to differ the options based on the form data. Previously I had this as a function on each option. This approach allows us to call a single function rather than one for each option.